src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -0,0 +1,576 @@
+                        {
+                            assetBuild = potentialBuild;
+                            break;
+                        }
+                    }
+                }
+
+                    
+                if (assetBuild == null)
+                {
+                    errors.Add($"Could not find build that produced {dependency.Name}@{dependency.Version} @ {dependency.Commit}");
+                    if (!_options.ContinueOnError)
+                    {
+                        break;
+                    }
+                }
+                else
+                {
+                    if (!builds.Contains(assetBuild))
+                    {
+                        builds.Add(assetBuild);
+                    }
+                }
+            }
+
+            List<Build> buildList = builds.ToList();
+
+            Console.WriteLine("Full set of builds in graph:");
+            foreach (var build in buildList)
+            {
+                Console.WriteLine($"  Build - {build.BuildNumber} of {build.Repository} @ {build.Commit}");
+            }
+
+            if (errors.Count != 0)
+            {
+                Console.WriteLine($"Failed to obtain full list of builds.  See errors:");
+                foreach (string error in errors)
+                {
+                    Console.WriteLine($" {error}");
+                }
+            }
+
+            return new Tuple<bool, List<Build>>(errors.Count == 0, builds.ToList()); ;
+        }
+
+        /// <summary>
+        /// Get an asset name that has the version (without it being double included)
+        /// </summary>
+        /// <param name="asset">Asset</param>
+        /// <returns>Name for logging.</returns>
+        private string GetAssetNameForLogging(Asset asset)
+        {
+            string assetNameAndVersion = asset.Name;
+            if (!assetNameAndVersion.Contains(asset.Version))
+            {
+                assetNameAndVersion += $"@{asset.Version}";
+            }
+            return assetNameAndVersion;
+        }
+
+        /// <summary>
+        ///     Gather the drop for a specific build.
+        /// </summary>
+        /// <param name="build">Build to gather drop for</param>
+        /// <param name="rootOutputDirectory">Output directory. Must exist.</param>
+        private async Task<bool> GatherDropForBuildAsync(Build build, string rootOutputDirectory)
+        {
+            DarcSettings darcSettings = LocalSettings.GetDarcSettings(_options, Logger);
+            // No need to set up a git type or PAT here.
+            Remote remote = new Remote(darcSettings, Logger);
+            bool success = true;
+
+            Console.WriteLine($"Gathering drop for build {build.BuildNumber} of {build.Repository}");
+            using (HttpClient client = new HttpClient())
+            {
+                var assets = await remote.GetAssetsAsync(buildId: build.Id, nonShipping: (!_options.IncludeNonShipping ? (bool?)false : null));
```

Comment:
>GetAssetsAsync [](start = 42, length = 14)

While I see you are doing the correct thing I think we can *probably* avoid this call, and hence make this a bit quicker, by reusing the call in line 220. I'm bringing this by looking only at the PR and not running the code or anything like that so probably I'm not seeing the reason behind having two calls. As well, I see tat in this we include nonShipping which might be the reason.

Reply:
One of these is to look up the specfic build an asset came from, and the other is to look up all assets from that build.